test/evp_test.c

```diff
@@ -2092,9 +2092,14 @@ static int kdf_test_init(EVP_TEST *t, const char *name)
 
     kdf = EVP_KDF_fetch(NULL, name, NULL);
     if (kdf == NULL) {
-        OPENSSL_free(kdata);
-        return 0;
+        /*
+         * Don't fail immediately. This algorithm might not be available in this
+         * provider. Keep going and see if we have an "Availablein" line. If
+         * not we will fail later.
+         */
```

Comment:
This is where the mem leak - I am happy for you to remove the test bits if it passes.